src/components/common/Icon/Icon.jsx

```diff
@@ -0,0 +1,16 @@
+import { memo, lazy, Suspense } from 'react';
+import dynamicIconImports from 'lucide-react/dynamicIconImports';
+
+const fallback = <div style={{ background: '#ddd', width: 20, height: 20 }} />;
+
+const Icon = ({ name, className, ...props }) => {
+  const LucideIcon = lazy(dynamicIconImports[name]);
+
+  return (
+    <Suspense fallback={fallback}>
+      <LucideIcon {...props} className={className} />
+    </Suspense>
+  );
+};
+
+export default memo(Icon);
```

Comment:
`withErrorBoundary` can be used here

Reply:
Or just wrap it into ErrorBoundary in the component?

Reply:
Yes, what you did in `DynamicIcon` is correct